tools/src/main/scala/scala/scalanative/codegen/CodeGen.scala

```diff
@@ -92,26 +87,32 @@ object CodeGen {
     private val deps             = mutable.Set.empty[Global]
     private val generated        = mutable.Set.empty[String]
     private val externSigMembers = mutable.Map.empty[Sig, Global.Member]
-    private val builder          = new ShowBuilder
-    import builder._
-
-    def gen(): ByteBuffer = {
-      genDefns(defns)
-      val body = builder.toString.getBytes("UTF-8")
-      builder.clear
-      genPrelude()
-      genConsts()
-      genDeps()
-      val prelude = builder.toString.getBytes("UTF-8")
-      val buffer  = ByteBuffer.allocate(prelude.length + body.length)
-      buffer.put(prelude)
-      buffer.put(body)
-    }
-
-    def genDeps() = deps.foreach { n =>
+
+    def gen(id: Option[Int], dir: VirtualDirectory): Unit = {
+      val body = dir.write(
+        Paths.get(s"${id.getOrElse("out")}-body.ll"),
+        writer => {
```

Comment:
```suggestion
        { writer =>
```

Reply:
I've applied it partially, because scalafmt enforces it we be in the same line with `Paths.get()`

Reply:
Ah yes. That makes sense. But in fact we could make this even better by declaring `write` using curried argument lists, so that we can call it with
```scala
val body = dir.write(Paths.get(s"${id.getOrElse("out")}-body.ll")) { writer =>
  ...
}
```